src/server_manager/web_app/app.ts

```diff
@@ -798,10 +798,12 @@ export class App {
     };
   }
 
-  private async setServerProgressView(server: server.Server): Promise<void> {
+  private async setServerProgressView(server: server.ManagedServer): Promise<void> {
     const view = await this.appRoot.getServerView(server.getId());
     view.serverName = this.makeDisplayName(server);
     view.selectedPage = 'progressView';
+    view.installProgress = server.installProgress();
+    server.onInstallProgressChange = progress => view.installProgress = progress;
```

Comment:
It's easier to understand if the listener lifecycle matches the server lifecycle.
This is better done where we load the servers. Replace where we do `await server.waitOnInstall();` with a loop that checks the progress, updates the view, and once progress = 1, moves on. Then you can delete `waitOnInstall()` and `onInstallCompleted()` and there will be no need for updating listeners.

Reply:
Polling loops are often easy to understand, but I think it's worth avoiding in this case.  Stacking multiple layers of polling (as would apply here) creates accumulating delay and overhead.  More concretely, `installProgress` can advance extremely fast (completing in less than one second) when loading a server that already exists (on startup/login).  A polling loop would either add a noticeable delay or need to poll extremely frequently.

Traditionally, JS has handled this situation with value-carrying events like [ProgressEvent](https://developer.mozilla.org/en-US/docs/Web/API/ProgressEvent).  However, that approach seems rather old-fashioned now.  If you're looking for a modern approach, the answer is probably to use an [async iterable](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of).  That's a bit unfamiliar in this codebase, but I'm happy to go in that direction if you would prefer.

Reply:
Changed to async-generator style.  Lots of fun!